lib/Model/FirewallRule.php

```diff
@@ -1,867 +1,493 @@
+            self::ACTION_ACCEPT,
+            self::ACTION_DROP,
+            self::ACTION_REJECT
+        ]);
+
+        $this->action = $action;
 
         return $this;
     }
 
     /**
-     * Gets protocol
-     * @return string
+     * @return string|float|null
      */
-    public function getProtocol()
+    public function getPosition()
     {
-        return $this->container['protocol'];
+        return $this->position;
     }
 
     /**
-     * Sets protocol
-     * @param string $protocol The protocol this rule will be applied to.
-     * @return $this
+     * @param string|float|null $position
+     * @return FirewallRule
      */
-    public function setProtocol($protocol)
+    public function setPosition($position): FirewallRule
     {
-        $allowed_values = $this->getProtocolAllowableValues();
-        if (!is_null($protocol) && !in_array($protocol, $allowed_values)) {
-            throw new \InvalidArgumentException(
-                sprintf(
-                    "Invalid value for 'protocol', must be one of '%s'",
-                    implode("', '", $allowed_values)
-                )
-            );
+        if (!is_null($position) && $position !== '') {
+            Assert::range($position, '1', '10000');
         }
-        $this->container['protocol'] = $protocol;
+
+        $this->position = $position;
 
         return $this;
     }
 
     /**
-     * Gets icmp_type
      * @return string
      */
-    public function getIcmpType()
+    public function getFamily(): string
     {
-        return $this->container['icmp_type'];
+        return $this->family;
     }
 
     /**
-     * Sets icmp_type
-     * @param string $icmp_type The ICMP type.
-     * @return $this
+     * @param string|null $family
+     * @return FirewallRule
      */
-    public function setIcmpType($icmp_type)
+    public function setFamily(?string $family): FirewallRule
     {
-        $this->container['icmp_type'] = $icmp_type;
+        Assert::oneOf($family, [
+            self::FAMILY_IP_V4,
+            self::FAMILY_IP_V6,
+        ]);
+
+        $this->family = $family;
 
         return $this;
     }
 
     /**
-     * Gets destination_address_start
-     * @return string
+     * @return string|null
      */
-    public function getDestinationAddressStart()
+    public function getProtocol(): ?string
     {
-        return $this->container['destination_address_start'];
+        return $this->protocol;
     }
 
     /**
-     * Sets destination_address_start
-     * @param string $destination_address_start The destination address range starts from this address.
-     * @return $this
+     * @param string|null $protocol
+     * @return FirewallRule
      */
-    public function setDestinationAddressStart($destination_address_start)
+    public function setProtocol(?string $protocol): FirewallRule
     {
-        $this->container['destination_address_start'] = $destination_address_start;
+        Assert::oneOf($protocol, [
```

Comment:
Protocol should be able to be set as null, since it's not required by the main API.

```
PHP Fatal error:  Uncaught InvalidArgumentException: Expected one of: "tcp", "udp", "icmp". Got: null in .../vendor/webmozart/assert/src/Assert.php:2042
Stack trace:
#0 /Users/trdls/upcloud-dev/apis/thirdparty-repos/devimasolutions/test-php-api/vendor/webmozart/assert/src/Assert.php(960): Webmozart\Assert\Assert::reportInvalidArgument('Expected one of...')
#1 /Users/trdls/upcloud-dev/apis/thirdparty-repos/devimasolutions/test-php-api/vendor/webmozart/assert/src/Assert.php(940): Webmozart\Assert\Assert::inArray(NULL, Array, '')
#2 /Users/trdls/upcloud-dev/apis/thirdparty-repos/devimasolutions/test-php-api/vendor/upcloudltd/upcloud-php-api/lib/Model/FirewallRule.php(241): Webmozart\Assert\Assert::oneOf(NULL, Array)
#3 /Users/trdls/upcloud-dev/apis/thirdparty-repos/devimasolutions/test-php-api/vendor/upcloudltd/upcloud-php-api/lib/Model/FirewallRule.php(113): Upcloud\ApiClient\Model\FirewallRule->setProtocol(NULL)
#4 /Users/trdls/upcloud-dev/apis/thirdparty-r in /Users/trdls/upcloud-dev/apis/thirdparty-repos/devimasolutions/test-php-api/vendor/webmozart/assert/src/Assert.php on line 2042
```

Reply:
@PopoSensei We've added one more commit. This issue should be resolved now.